components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/TokenManagementDAOImpl.java

```diff
@@ -161,18 +180,13 @@ public RefreshTokenValidationDataDO validateRefreshToken(String consumerKey, Str
                     validationDataDO.setTokenId(resultSet.getString(9));
                     validationDataDO.setGrantType(resultSet.getString(10));
                     String subjectIdentifier = resultSet.getString(11);
-                    AuthenticatedUser user = OAuth2Util.createAuthenticatedUser(userName, userDomain, tenantDomain);
-                    user.setAuthenticatedSubjectIdentifier(subjectIdentifier);
-
-                    if (!OAuthServerConfiguration.getInstance().isMapFederatedUsersToLocal() && userDomain.startsWith
-                            (OAuthConstants.UserType.FEDERATED_USER_DOMAIN_PREFIX)) {
-                        if (log.isDebugEnabled()) {
-                            log.debug("Federated prefix found in domain " + userDomain + " and " +
-                                    "federated users are not mapped to local users. " +
-                                    "Hence setting user to a federated user for client id" + consumerKey);
-                        }
-                        user.setFederatedUser(true);
+                    String authenticatedIDP = null;
+                    if (OAuth2ServiceComponentHolder.isIDPIdColumnEnabled()) {
+                        authenticatedIDP = resultSet.getString(12);
                     }
+                    AuthenticatedUser user = OAuth2Util.createAuthenticatedUser(userName, userDomain, tenantDomain,
+                            authenticatedIDP);
+                    user.setAuthenticatedSubjectIdentifier(subjectIdentifier);
```

Comment:
This is the only place I noted that subjectIdentifier is being set explicitly.
It should be set in each case right as being retrieved from the DB or the same create method should handle it

Reply:
This line of code was actually previously [there](https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1097/files/ba51e19b7c89ebd9e8a105186974e25b284498eb#diff-88d0d684bd0bd2fb1bc1b5bd8f26f48cL165). There are also several other places where this is being used.